### DISCUSSION TITLE
Reject nested keys

### DISCUSSION THREAD

**Discussion by tdaniely-dn** [nushell/nushell#6408] [Q&A] [answered]
New to this so trying to figure stuff out.
I want to filter out nested fields at specific paths from json data.

After much playing around I found that the naive reject approach doesn't work.
Wondering if this is a bug, limitation or somehow by design.
To me the reject syntax here is desirable.

With this approach, to do a complex mutation on deep structured data, I reason that I need a set of typed visitors.
Does this make sense?
Is there a way to morph the data into a key-path -> value map and back?

I know it can be done with other tools, just investigating nu a bit.

```nu
❯ let x = {a:{b:3,c:5}}

# works
❯ $x | update a.b 5 | debug
{a: {b: 5, c: 5}}

# doesn't work
❯ $x | reject a.b | debug
{a: {b: 3, c: 5}}

# workaround
❯ $x | update a { |x| $x.a | reject b } | debug
{a: {c: 5}}
```

**Comment by fdncred** [comment, marked as answer]
I agree, it would be nice if `reject` worked with deeper column path "addresses". We have a few commands that would benefit from this type of then. Anyone who wants to manipulate deeply nested json has this problem unfortunately. I can't answer your remaining questions. Perhaps others will chime in.

**Comment by tdaniely-dn** [comment]
After some more poking around endede up with this filler.

```nu
def reject_deep [p: cell-path, f: cell-path] {
    update $p { |x| $x | get $p | reject $f }
}
```

**Reply by fdncred** [reply]
nice work!
